File: src/test/java/uk/gov/justice/digital/hmpps/oauth2server/timed/DisableInactiveAuthUsersTest.java
package uk.gov.justice.digital.hmpps.oauth2server.timed;

import com.microsoft.applicationinsights.TelemetryClient;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.junit.MockitoJUnitRunner;

import static org.mockito.Mockito.verify;
import static org.mockito.Mockito.when;

@RunWith(MockitoJUnitRunner.class)
public class DisableInactiveAuthUsersTest {
    @Mock
    private DisableInactiveAuthUsersService service;
    @Mock
    private TelemetryClient telemetryClient;

    private DisableInactiveAuthUsers disableInactiveAuthUsers;

    @Before
    public void setUp() {
        disableInactiveAuthUsers = new DisableInactiveAuthUsers(service, telemetryClient);
    }

    @Test
    public void findAndDisableInactiveAuthUsers() {
        when(service.processInBatches()).thenReturn(0);
        disableInactiveAuthUsers.findAndDisableInactiveAuthUsers();
        verify(service).processInBatches();
    }
}

File: src/test/kotlin/uk/gov/justice/digital/hmpps/oauth2server/timed/DisableInactiveAuthUsersTest.kt
package uk.gov.justice.digital.hmpps.oauth2server.timed

import com.microsoft.applicationinsights.TelemetryClient
import org.junit.jupiter.api.Test
import org.mockito.kotlin.mock
import org.mockito.kotlin.verify
import org.mockito.kotlin.whenever

class DisableInactiveAuthUsersTest {
  private val service: DisableInactiveAuthUsersService = mock()
  private val telemetryClient: TelemetryClient = mock()
  private var disableInactiveAuthUsers = DisableInactiveAuthUsers(service, telemetryClient)

  @Test
  fun findAndDisableInactiveAuthUsers() {
    whenever(service.processInBatches()).thenReturn(0)
    disableInactiveAuthUsers.findAndDisableInactiveAuthUsers()
    verify(service).processInBatches()
  }
}
